pkg/tsdb/azuremonitor/azure-log-analytics-datasource.go

```
@@ -248,153 +276,29 @@ func (e *AzureLogAnalyticsDatasource) unmarshalResponse(res *http.Response) (Azu
-				}
-
-				if valueIndex == -1 && (v.Type == "int" || v.Type == "long" || v.Type == "real" || v.Type == "double") {
-					valueIndex = i
-				}
-			}
-
-			if timeIndex == -1 {
-				azlog.Info("No time column specified. Returning existing columns, no data")
-				return nil, simplejson.NewFromAny(meta), nil
-			}
-
-			if valueIndex == -1 {
-				azlog.Info("No value column specified. Returning existing columns, no data")
-				return nil, simplejson.NewFromAny(meta), nil
-			}
-
-			slice := tsdb.TimeSeriesSlice{}
-			buckets := map[string]*tsdb.TimeSeriesPoints{}
-
-			getSeriesBucket := func(metricName string) *tsdb.TimeSeriesPoints {
-				if points, ok := buckets[metricName]; ok {
-					return points
-				}
-
-				series := tsdb.NewTimeSeries(metricName, []tsdb.TimePoint{})
-				slice = append(slice, series)
-				buckets[metricName] = &series.Points
-
-				return &series.Points
-			}
-
-			for _, r := range t.Rows {
-				timeStr, ok := r[timeIndex].(string)
-				if !ok {
-					return nil, simplejson.NewFromAny(meta), errors.New("invalid time value")
-				}
-				timeValue, err := time.Parse(time.RFC3339Nano, timeStr)
-				if err != nil {
-					return nil, simplejson.NewFromAny(meta), err
-				}
-
-				var value float64
-				if value, err = getFloat(r[valueIndex]); err != nil {
-					return nil, simplejson.NewFromAny(meta), err
-				}
-
-				var metricName string
-				if metricIndex == -1 {
-					metricName = t.Columns[valueIndex].Name
-				} else {
-					metricName, ok = r[metricIndex].(string)
-					if !ok {
-						return nil, simplejson.NewFromAny(meta), err
-					}
-				}
-
-				points := getSeriesBucket(metricName)
-				*points = append(*points, tsdb.NewTimePoint(null.FloatFrom(value), float64(timeValue.Unix()*1000)))
-			}
-
-			return slice, simplejson.NewFromAny(meta), nil
-		}
-	}
-
-	return nil, nil, errors.New("no data as no PrimaryResult table was returned in the response")
+	azlog.Error("failed to encode the query into the encodedQuery property")
```

Maybe want to include the error? Do we want to return this error or what is the query result if this error happens?

---

I don't think it is a very likely error so I left it as a log. And since it is metadata, I expect there would still be a query result that we wouldn't want to fail because of this. I could perhaps add a notice to the frame like do with long to wide, although how that is used in Grafana I think is still not well defined.

---

Okay make sense.